src/metabase/embed/settings.clj

```diff
@@ -58,3 +60,22 @@
   :default    false
   :export?    true
   :visibility :admin)
+
+(defsetting enable-embedding-sdk
+  (deferred-tru "Used for enabling or disabling embedding SDK.")
+  :type       :boolean
+  :default    false
+  :visibility :authenticated
+  :export?    true
+  :audit      :getter
+  :setter     (fn [new-value]
+                (when (not= new-value (setting/get-value-of-type :boolean :enable-embedding-sdk))
+                  (setting/set-value-of-type! :boolean :enable-embedding-sdk new-value)
+                  (when (and new-value (str/blank? (embed/embedding-secret-key)))
```

Comment:
Does this need the `new-value` in the and? 
If I understand, it will create a new secret key once, and then never again, even if we toggle new-value true/false.

Reply:
I think you're right that it's not needed. I do want to match the logic in https://github.com/metabase/metabase/pull/47637/files#diff-d57e6bb846a5ba607fb0b7125e76c152158a4ba15becff5650b80359e256faaeR33

So, I think I'll change that in a followup after we make the switch. I want this to be drop-in replacement as much as possible